### PR TITLE
Add feature to manage markers

### PR DESCRIPTION
**Pull request by mmouchous-ledger**
Gives the possibility for the user to 
- display markers as a list grouped by their color (it may be a future feature to give them a group name).
- controls to show/hide markers in the viewer

### CODE REVIEW

File: laserstudio/widgets/toolbars/markerslistdialog.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        super().__init__(parent)
+        self._number_of_checked = 0
+
+    @property
+    def number_of_checked(self):
+        return self._number_of_checked
+    
+    @number_of_checked.setter
+    def number_of_checked(self, value):
+        self._number_of_checked = value
```
Comment (kingofpayne) [inline laserstudio/widgets/toolbars/markerslistdialog.py:18]: There is currently no benefit in having this property, you can simplify the code by removing the getter and setter and renaming `_number_of_checked` to `number_of_checked`.

Comment (mmouchous-ledger) [reply laserstudio/widgets/toolbars/markerslistdialog.py]: It was the first intention, but it has been revised afterwards

File: laserstudio/widgets/toolbars/markerslistdialog.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+class MarkerListItem(QTreeWidgetItem):
+    def __init__(self, group: MarkerGroupListItem, marker: Marker):
+        super().__init__(group)
+        self.group = group
+        x, y  = marker.pos().x(), marker.pos().y()
+        self.marker = marker
+        visible = marker.isVisible()
+        self.setCheckState(0, Qt.CheckState.Checked if visible else Qt.CheckState.Unchecked)
+        self.setText(0, f"{x:.02f}µm {y:.02f}µm")
```
[the remark this answers is not in the document]
Comment (mmouchous-ledger) [reply laserstudio/widgets/toolbars/markerslistdialog.py]: Added there and some other places in laserstudio

File: laserstudio/laserstudio.py [hunk 3]
```diff
@@ -185,11 +186,11 @@ def __init__(self, config: Optional[dict]):
         if window_state is not None:
             self.restoreState(window_state)
 
-    def closeEvent(self, event):
+    def closeEvent(self, a0):
```
Comment (kingofpayne) [inline laserstudio/laserstudio.py:189]: search/replace mistake?

Comment (mmouchous-ledger) [reply laserstudio/laserstudio.py:189]: No, for pyQt6, QtWidgets.pyi contains "def closeEvent(self, a0: typing.Optional[QtGui.QCloseEvent]) -> None: ...", so PyLance was complaining that the redefinition of the function does not have the same parameters name.

**Pull request merged**